astropy/coordinates/baseframe.py

```diff
@@ -1156,8 +1156,17 @@ def transform_to(self, new_frame):
             # likely a SkyCoord object.
             new_frame = new_frame._sky_coord_frame
 
+        if isinstance(new_frame, str):
```

Comment:
This already exists as `sky_coordinate._get_frame_class`, I don't think we should duplicate it; I think it may make most sense to move it to `baseframe` (together with `_get_repr_cls`) and then import it in `sky_coordinate` (not quite OK with a private function, but duplication is worse).

But let me ping @eteq, since he has been thinking about all those functions in `SkyCoord`.